tracepoint_perf/examples/rewrite_perf.rs

```diff
@@ -172,17 +172,31 @@ fn main() -> process::ExitCode {
         // already populated, and we merge all descriptors with names before merging any
         // descriptors that don't have names.
 
+        let mut merge_event_scratch_buffer = vec::Vec::new();
+
         // Prefer data from descriptors that have names.
         for desc in input.event_desc_list() {
             if !desc.name().is_empty() {
-                merge_event_desc(&mut output, &output_path, &mut sample_ids_used, desc);
+                merge_event_desc(
```

Comment:
Note: Allow `merge_event_desc` to reuse the buffer instead of allocating and freeing a buffer each time it is called.